syntax/crisp_lexer.mll

```diff
@@ -19,49 +53,38 @@ Stack.push Crisp_syntax.min_indentation scope_stack;;
 let test_indentation indentation follow_on_tokens lexbuf =
   assert (not (Stack.is_empty scope_stack)); (*There should always be at least
                                                one element in the stack: 0*)
-  let next_line () =
-    (*this function was adapted from
-      https://realworldocaml.org/v1/en/html/parsing-with-ocamllex-and-menhir.html*)
-    let pos = lexbuf.lex_curr_p in
-    lexbuf.lex_curr_p <-
-      { pos with pos_bol = lexbuf.lex_curr_pos;
-        pos_lnum = pos.pos_lnum + 1
-      } in
   (*Count how many scopes we've moved down (out of).*)
   let rec undented_scopes (offset : int) =
     if Stack.top scope_stack = indentation then
       offset
     else if Stack.top scope_stack < indentation then
-      failwith "Undershot the scope?"
+      raise Error
     else
       begin
         ignore(Stack.pop scope_stack);
         undented_scopes (offset + 1)
       end in
   let prev = Stack.top scope_stack in
-    if indentation > prev then
-      begin
-        Stack.push indentation scope_stack;
-        next_line ();
-        INDENT
-      end
-    else if indentation = prev then
-      begin
-        next_line ();
-        if follow_on_tokens <> [] then
-          begin
-            assert (List.length follow_on_tokens = 1);
-            List.hd follow_on_tokens
-          end
-        else NL
-      end
-    else
-      begin
-        assert (indentation < prev);
-        next_line ();
-        UNDENTN (undented_scopes Crisp_syntax.min_indentation,
-                 follow_on_tokens)
-      end
+  if indentation > prev then
```

Comment:
I'm currently using [VS Code](https://code.visualstudio.com/) with the OCaml plugin, which integrates ocaml-merlin, ocp-indent, and ocamldebug. The only annoyance is it doesn't add the Unix trailing \n!